client/components/backup-retention-management/retention-options/retention-option-card.tsx

```diff
@@ -0,0 +1,75 @@
+import classnames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import React from 'react';
+import type { TranslateResult } from 'i18n-calypso';
+
+interface RetentionOptionCardProps {
+	label: string;
+	spaceNeeded: TranslateResult | string;
+	upgradeRequired: boolean;
+	isCurrentPlan: boolean;
+	value: number;
+	checked?: boolean;
+	onChange: ( value: number ) => void;
+}
+
+const RetentionOptionCard: React.FC< RetentionOptionCardProps > = ( {
+	label,
+	spaceNeeded,
+	upgradeRequired,
+	isCurrentPlan,
+	value,
+	checked = false,
+	onChange,
+} ) => {
+	const translate = useTranslate();
+
+	const handleOptionClick = () => {
+		onChange( value );
+	};
+
+	return (
+		<div
+			className="retention-option"
+			role="radio"
+			aria-checked={ checked }
+			onClick={ handleOptionClick }
+			onKeyDown={ handleOptionClick }
+			tabIndex={ 0 }
+		>
```

Comment:
Any more accessible way to compose a radio button? We have a bunch of radio control components available for use.